ambry-account/src/test/java/com/github/ambry/account/InMemoryUnknownAccountServiceTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import com.github.ambry.utils.Utils;
+import com.github.ambry.utils.UtilsTest;
+import java.util.Collections;
+import java.util.Random;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Unit tests for {@link InMemoryUnknownAccountService} and {@link InMemoryUnknownAccountServiceFactory}.
+ */
+public class InMemoryUnknownAccountServiceTest {
+  private static final Random random = new Random();
+  private AccountService accountService;
+
+  @After
+  public void close() throws Exception {
+    accountService.close();
+  }
+
+  @Test
+  public void testAllMethods() throws Exception {
+    accountService = new InMemoryUnknownAccountServiceFactory(null, null, null).getAccountService();
+    Account refAccount = new AccountBuilder((short) 1, "a", Account.AccountStatus.INACTIVE, null).build();
```

Comment:
this is not really a "refAccount" (i.e. not a reference) - can we name it appropriately and declare it where it is required (not this early).